### DISCUSSION TITLE
Looking for lodash(-es) replacements

### DISCUSSION THREAD

**Discussion by fabianwohlfart** [e18e/e18e#9] [Q&A] [answered]
Hi guys, any advice on replacements for `debounce`,` throttle`, `set`, `get`, `defaults` from `lodash`?
I think for debounce and throttle i'd just go for `usedebouncefn` and `usethrottlefn` from @vueuse, `defaults` is just spread operator, right?, but `set` and `get`? 
Any leads welcome.

**Comment by SukkaW** [comment]
`get` you can just use [the optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

Before:

```ts
get(nullable.nullable.optionalField)
```

After:

```ts
nullable?.nullable?.optionalField
```

And optional chaining is typescript-friendly!

**Reply by fabianwohlfart** [reply]
Would need to modify and save the path, so optional chaining is a bit to little for my use case. Thanks!

**Comment by zloirock** [comment, marked as answer]
https://you-dont-need.github.io/You-Dont-Need-Lodash-Underscore/

**Reply by fabianwohlfart** [reply]
Also found this now, they share the same idea, but have different functions. https://youmightnotneed.com/lodash

**Comment by sebakerckhof** [comment]
Does it actually help to replace lodash? If it can be replaced by a built-in feature, sure. But for other utils?

If you use lodash-es, a modern bundler can tree-shake so you're only including the necessary utils. So if instead every package now starts to implement their own alternative debounce implementation or all of them start to use a bunch of other packages, is the end result not worse?

Situation before: 100 of my dependencies depend on lodash
Situation after: 80 of my dependencies depend on lodash, The other 20 now have 17 custom implementations of some lodash functions and use 35 different little packages to replace lodash functions.

**Reply by 43081j** [reply]
I don't think it would actually turn out that extreme

Situation before: 100 of your dependencies depend on multiple version of lodash

Situation after: 80 depend on lodash, the other 20 use the platform or a common/popular smaller library

Most lodash uses get replaced by native calls. Stuff like debounce I expect would have a popular library or two we all depend on (which is what naturally happens with all dependencies)
